spikeextractors/recordingextractor.py

```diff
@@ -496,11 +496,20 @@ def copy_channel_properties(self, recording, channel_ids=None):
         else:
             if isinstance(channel_ids, (int, np.integer)):
                 channel_ids = [channel_ids]
+
+            # copy key properties
+            groups = recording.get_channel_groups(channel_ids=channel_ids)
+            locations = recording.get_channel_locations(channel_ids=channel_ids)
+            self.set_channel_groups(groups)
+            self.set_channel_locations(locations)
+
+            # copy normal properties
             for channel_id in channel_ids:
                 curr_property_names = recording.get_channel_property_names(channel_id=channel_id)
                 for curr_property_name in curr_property_names:
-                    value = recording.get_channel_property(channel_id=channel_id, property_name=curr_property_name)
-                    self.set_channel_property(channel_id=channel_id, property_name=curr_property_name, value=value)
+                    if curr_property_name not in self._key_properties.keys():  # key property
+                        value = recording.get_channel_property(channel_id=channel_id, property_name=curr_property_name)
+                        self.set_channel_property(channel_id=channel_id, property_name=curr_property_name, value=value)
```

Comment:
This is ok for the moment this should change when implementing the dict of np.array approach